2.evaluate_data/class_balanced_log_reg_areashape_predict_sc_probabilities/nbconverted/class_balanced_log_reg_areashape_predict_sc_probabilities.py

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Predict Single Cell Probabilities
+# In this part of the analysis, the predicted single cell probabilities are stored in parquet files per plate.
+# Predicted probabilities are generated from a weighted logistic regression trained on AreaShape features from the mitocheck morphology features.
+
+# ### Import Libraries
+
+# In[ ]:
+
+
+import gzip
+import io
+import pathlib
+import pandas as pd
+import requests
+from joblib import load
+import sys
+
+
+# ## Find the path of the git directory
+
+# In[ ]:
+
+
+# Get the current working directory
+cwd = pathlib.Path.cwd()
+
+if (cwd / ".git").is_dir():
+    root_dir = cwd
+
+else:
+    root_dir = None
+    for parent in cwd.parents:
+        if (parent / ".git").is_dir():
+            root_dir = parent
+            break
+
+# Check if a Git root directory was found
+if root_dir is None:
+    raise FileNotFoundError("No Git root directory found.")
+
+
+# ## Load data code
+
+# In[ ]:
+
+
+def load_joblib_from_url(url):
```

Comment:
I like this function - very clever

Reply:
Thanks, it was Greg's idea from a long time ago